lib/grover/js/processor.cjs

```diff
@@ -1,4 +1,42 @@
-// Setup imports
+const fs = require('fs')
+const path = require('path')
+
+function findPnpFile(startDir) {
+  let currentDir = startDir;
+  while (currentDir !== path.parse(currentDir).root) {
+      const pnpPath = path.join(currentDir, '.pnp.cjs');
+      if (fs.existsSync(pnpPath)) {
```

Comment:
although the likelihood of this being a TOCTOU issue is low, I think it'd still be worth refactoring to avoid using any `exists` methods. ie 
1. calculate the path (the line above)
2. use it by calling `require(pnpPath).setup();` (wrapped in a try/catch)
   - if it succeeds, well.. you know the path was good and we can just try require puppeteer/core
   - if it didn't succeed then we try the next path

It would require flipping the logic a little on its head, but something like this would appear to do it. Note, I don't have yarn 4 on my system so I haven't tested this 😬. Speaking of testing, it would be good if there was a way for this to be tested as a part of the CI so as to avoid regressions. I'm keen to hear what thoughts you might have on this 😄 

```
function checkPuppeteerInstalledWithPnp() {
  let currentDir = process.cwd();
  while (currentDir !== path.parse(currentDir).root) {
    const pnpPath = path.join(currentDir, '.pnp.cjs');
    try {
      require(pnpPath).setup();
      try {
        require('puppeteer');
      } catch (error) {
        require('puppeteer-core');
      }
      return;
    } catch (loadError) {
      // That didn't work.. lets try the next path
    }
  }
  throw('Error: .pnp.cjs file not found.');
}
```

Reply:
will look into this in a few days... on vacation for the last few weeks... also worth noting i'm super new to any JS stuff, so I'm learning here!  thanks for the feedback and I plan to address when I'm back home :pray:

Reply:
no worries, enjoy the holiday 😄